Mathlib/CategoryTheory/ObjectProperty/ColimitsOfShape.lean

```diff
@@ -182,6 +194,159 @@ lemma prop_colimit (F : J ⥤ C) [HasColimit F] (hF : ∀ (j : J), P (F.obj j))
 
 end
 
+variable {J} in
+lemma colimitsOfShape_le_of_final (G : J ⥤ J') [G.Final] :
+    P.colimitsOfShape J' ≤ P.colimitsOfShape J :=
+  fun _h ⟨h⟩ ↦ ⟨h.reindex G⟩
+
+variable {J} in
+lemma colimitsOfShape_congr (e : J ≌ J') :
+    P.colimitsOfShape J = P.colimitsOfShape J' :=
+  le_antisymm (P.colimitsOfShape_le_of_final e.inverse)
+    (P.colimitsOfShape_le_of_final e.functor)
+
+variable {J} in
+lemma isClosedUnderColimitsOfShape_iff_of_equivalence (e : J ≌ J') :
+    P.IsClosedUnderColimitsOfShape J ↔
+      P.IsClosedUnderColimitsOfShape J' := by
+  simp only [isClosedUnderColimitsOfShape_iff, P.colimitsOfShape_congr e]
+
+variable {P J} in
+lemma IsClosedUnderColimitsOfShape.of_equivalence (e : J ≌ J')
+    [P.IsClosedUnderColimitsOfShape J] :
+    P.IsClosedUnderColimitsOfShape J' := by
+  rwa [← P.isClosedUnderColimitsOfShape_iff_of_equivalence e]
+
+lemma colimitsOfShape_eq_unop_limitsOfShape :
+    P.colimitsOfShape J = (P.op.limitsOfShape Jᵒᵖ).unop := by
+  ext X
+  constructor
+  · rintro ⟨h⟩
+    exact ⟨{
+      diag := h.diag.op
+      π := NatTrans.op h.ι
+      isLimit := isLimitOfUnop h.isColimit
+      prop_diag_obj _ := h.prop_diag_obj _
+    }⟩
+  · rintro ⟨h⟩
+    exact ⟨{
+      diag := h.diag.unop
+      ι := NatTrans.unop h.π
+      isColimit := isColimitOfOp h.isLimit
+      prop_diag_obj _ := h.prop_diag_obj _
+    }⟩
```

Comment:
```suggestion
  refine ⟨fun ⟨h⟩ => ⟨?_⟩, fun ⟨h⟩ => ⟨?_⟩⟩
  · exact
      { diag := h.diag.op
        π := NatTrans.op h.ι
        isLimit := isLimitOfUnop h.isColimit
        prop_diag_obj _ := h.prop_diag_obj _ }
  · exact
      { diag := h.diag.unop
        ι := NatTrans.unop h.π
        isColimit := isColimitOfOp h.isLimit
        prop_diag_obj _ := h.prop_diag_obj _ }
```

Using this `refine ⟨fun ⟨h⟩ => ⟨?_⟩, fun ⟨h⟩ => ⟨?_⟩⟩` instead of `constructor` trick, we can remove the dreaded `⟨{` that forces orphaned delimiters.

It is, however purely a matter of taste, so completely up to you.